### PR TITLE
Adds ability to seek with arrow keys

### PR DESCRIPTION
**Pull request by jacksongoode**
This changes the arrow key behavior:
- Left/right - seek 10s forward/behind
- Shift+left/right - Previous/next track

### CODE REVIEW

File: psst-gui/src/ui/playback.rs [hunk 1]
```diff
@@ -90,29 +90,27 @@ fn playing_item_widget() -> impl Widget<NowPlaying> {
...
+                    .context_menu(|now_playing| match &now_playing.item {
+                        Playable::Track(track) => {
+                            track::track_menu(track, &now_playing.library, &now_playing.origin)
+                        }
+                        Playable::Episode(episode) => {
+                            episode::episode_menu(episode, &now_playing.library)
+                        }
+                    }),
+                1.0,
+            ),
```
Comment (jacksongoode) [inline psst-gui/src/ui/playback.rs:113]: This was a linting format.

**Comment by jacksongoode** [comment]
@Insprill If you have some time, let me know if this new functionality looks good to you.

File: psst-gui/src/controller/playback.rs [hunk 2]
```diff
@@ -412,11 +412,25 @@ where
...
             }
             Event::KeyDown(key) if key.code == Code::ArrowRight => {
-                self.next();
+                if key.mods.shift() {
+                    self.next();
+                } else if let Some(now_playing) = &data.playback.now_playing {
+                    let current_position = now_playing.progress;
+                    let max_position = now_playing.item.duration();
+                    let seek_position =
+                        (current_position + Duration::from_secs(10)).min(max_position);
```
Comment (Insprill) [inline psst-gui/src/controller/playback.rs:421]: Is there a particular reason this is implemented differently when going forward vs. backward?

Comment (Insprill) [inline psst-gui/src/controller/playback.rs:421]: We should probably make the skip duration configurable - I can see many people wanting that.

**Review by Insprill — CHANGES_REQUESTED** [review]

File: psst-gui/src/controller/playback.rs (hunk 2 above)
Comment (jacksongoode) [reply psst-gui/src/controller/playback.rs:421]: Good idea!

Comment (jacksongoode) [reply psst-gui/src/controller/playback.rs]: Noted, recent commit standardizes this.

Comment (jacksongoode) [reply psst-gui/src/controller/playback.rs]: Implemented in the recent commit.

File: psst-gui/src/ui/preferences.rs [hunk 3]
```diff
@@ -212,6 +212,22 @@ fn general_tab_widget() -> impl Widget<AppState> {
 
     col = col.with_spacer(theme::grid(3.0));
 
+    col = col
+        .with_child(Label::new("Step Duration").with_font(theme::UI_FONT_MEDIUM))
```
Comment (Insprill) [inline psst-gui/src/ui/preferences.rs:216]: I think this should be called "Skip Duration" or "Seek Duration"

Comment (jacksongoode) [reply psst-gui/src/ui/preferences.rs:216]: Good call

**Review by Insprill — APPROVED** [review]
LGTM!

**Pull request merged**